### DISCUSSION TITLE
Loading almost 3000 SVG files when opening web page

### DISCUSSION THREAD

**Discussion by SimonVallieresPMG** [fomantic/Fomantic-UI#1977] [General] [answered]
Hi,

I am a developer in a small company. We are currently entirely on Semantic-UI. For some time now, I discovered Fomantic-UI and wanted to give it a try. (We work with Spring Boot, Thymeleaf, WebJars, HTML, JS and CSS)

To test your library, I simply made a clone of the repos and pointed my JS and CSS dependencies from Semantic-UI to your library.

(the folder called "fomantic-ui" is the "dist" folder of the repo)

```
<!--<link rel="stylesheet" th:href="@{/webjars/Semantic-UI/2.4.1/semantic.min.css}"/>-->
<link rel="stylesheet" th:href="@{/fomantic-ui/semantic.min.css}"/>
<!--<script th:src="@{/webjars/Semantic-UI/2.4.1/semantic.min.js}"></script>-->
<script th:src="@{/fomantic-ui/semantic.min.js}"></script>
```

### Everything works perfectly except for a big detail. Every time I load a web page, I have almost 3000 svg files to download.

![fomantic-ui-Network-tab](https://user-images.githubusercontent.com/85201078/120499176-d8cc3a00-c38d-11eb-9899-0d8f390dd2bd.gif)

So here is my questions:

### If that's not possible, how do you prevent it from slowing down the loading of a page?

**Comment by lubber-de** [comment, marked as answer]
Including the scripts alone does not trigger all those requests. So there seems to be something else on your page.
Could you please create a jsfiddle, so we can reproduce your behavior?

Your log screenshots show that index.js is loading all those SVG Files and they are inline data streams, which we don't have. ONly the emoji component uses svg files but those are linked to a cdn.

**Reply by SimonVallieresPMG** [reply]
You are absolutely right.

Apparently, all of these images are loaded because of one of my extensions. [Dark Reader] (https://chrome.google.com/webstore/detail/dark-reader/eimadpbcbfnmbkopoojfekhnkhdbieeh?hl=en).

However, I still haven't found a way to solve this problem. On the other hand, this problem also occurs on the site https://fomantic-ui.com/ and only when the mode of DarkReader is set to "dynamic".

![image](https://user-images.githubusercontent.com/85201078/120512747-a0325d80-c399-11eb-8ba2-1c37700b0e80.png)

It is therefore obvious that this problem does not concern Fomantic-UI directly, but DarkReader.

Actually what I'm guessing is darkreader is trying to convert all Fomantic-UI svg images to "dark mode". Which consequently charges all of them despite the fact that they are not used on the web page.

[Here is a link to the issue I just created on DarkReader Git](https://github.com/darkreader/darkreader/issues/5942)

Thank you!
